tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11430,12 +11430,28 @@ void ValidatePatchConstantFunctionsExist(clang::Sema *self) {
 // should exist on this shader stage
 void DiagnoseEntryAttrAllowedOnStage(clang::Sema *self,
                                      FunctionDecl *entryPointDecl,
-                                     DXIL::ShaderKind shaderKind) {
+                                     DXIL::ShaderKind shaderKind,
+                                     bool isActiveEntry) {
 
   if (entryPointDecl->hasAttrs()) {
     for (Attr *pAttr : entryPointDecl->getAttrs()) {
       switch (pAttr->getKind()) {
-
+      // if this is a "node" shader and the target stage isn't compute nor
+      // library, emit a diagnostic
```

Comment:
No, we don't support node shader entry points compiled to a compute shader target.

Note: "library" isn't a shader "stage".  It may contain entry points for various shader stages, but it isn't a stage itself.

In `DiagnoseEntryAttrAllowedOnStage`, `shaderKind` comes from the attribute on the function.  Either this attribute was provided by the user, or it was added to the function if you're compiling directly to a non-library target and there's no existing or conflicting attribute already present.

If you try to compile a node shader to a compute target, code in `TryAddShaderAttrFromTargetProfile` should already have emitted a diagnostic that the profile conflicts with the shader attribute on the function.  Note that the comment above that diagnostic is incorrect: the two lines talking about a special case for compute+node should be removed.

No conflicting attribute should exist at this point, because that would have been caught earlier.

Therefore, there is no need to do anything with the `HLSLShader` attribute here.